src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx

```diff
@@ -71,7 +73,12 @@ function BaseOnboardingPurpose({shouldUseNativeStyles, shouldEnableMaxHeight, ro
                 Welcome.setOnboardingErrorMessage('');
 
                 if (choice === CONST.ONBOARDING_CHOICES.MANAGE_TEAM) {
-                    Navigation.navigate(ROUTES.ONBOARDING_WORK.getRoute(route.params?.backTo));
+                    if (!onboardingPolicyID) {
+                        const {adminsChatReportID, policyID} = Policy.createWorkspace(undefined, true);
+                        Welcome.setOnboardingAdminsChatReportID(adminsChatReportID);
+                        Welcome.setOnboardingPolicyID(policyID);
+                    }
```

Comment:
Since we are not going to use the policy in between steps is there any reason that we create the workspace at that early step? Why not create it just before calling `completeOnboarding` in the accounting step?

Reply:
Interesting. Did you have any concern about that or opinion on why it would be better to do one over the other?

Reply:
No strong preference it's just a little questionable that we create the workspace when not really needed yet. Now I think about it more I think we may have a bug:
1. Login as a new user
2. On the onboarding flow, select purpose to manage team <-- This will create a workspace
3. Logout
4. Login in again
5. If you try to complete the onboarding flow you will be creating another workspace since `onboardingPolicyID` is not on onyx

cc @nkdengineer Can you please confirm if you can reproduce this bug?

Reply:
Oh hmm that's an interesting edge case for sure - I'm not sure how we would completely solve that without designating a specific policy as the "onboarding policy". Though we could potentially check if a user has an existing policy.